WalletWasabi/Blockchain/Keys/HdPubKey.cs

```diff
@@ -135,6 +138,16 @@ public void SetKeyState(KeyState state, KeyManager? kmToFile = null)
 
 		kmToFile?.ToFile();
 	}
+	
+	public void SetObsoleteHeight(Height height)
+	{
+		if (ObsoleteHeight == height)
+		{
+			return;
+		}
+
+		ObsoleteHeight = height;
+	}
```

Comment:
This method feels redundant.

Even if `ObsoleteHeight == height` are equal, nothing wrong with assign it again.

Also, `ObsoleteHeight` is public, so why not just use the property straight away?